src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibCompressionOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO.Compression
+{
+    /// <summary>
+    /// Provides compression options to be used with <see cref="ZLibStream"/>, <see cref="DeflateStream"/> and <see cref="GZipStream"/>.
+    /// </summary>
+    public sealed class ZLibCompressionOptions
+    {
+        private int _compressionLevel = -1;
+        private ZLibCompressionStrategy _strategy;
+
+        /// <summary>
+        /// Gets or sets the compression level for a compression stream.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the value is less than -1 or greater than 9.</exception>"
+        /// <remarks>
+        /// Can accept any value between -1 and 9 (inclusive), 0 gives no compression,  1 gives best speed, 9 gives best compression.
+        /// and -1 requests the default compression level which is currently equivalent to 6.
+        /// </remarks>
+        public int CompressionLevel
+        {
+            get => _compressionLevel;
+            set
+            {
+                ArgumentOutOfRangeException.ThrowIfLessThan(value, -1);
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(value, 9);
+
+                _compressionLevel = value;
+            }
+        }
+        /// <summary>
+        /// Gets or sets the compression algorithm for a compression stream.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException" >Thrown when the value is not a valid <see cref="ZLibCompressionStrategy"/> value.</exception>"
+        public ZLibCompressionStrategy CompressionStrategy
+        {
+            get => _strategy;
+            set
+            {
+                if (value < ZLibCompressionStrategy.Default || value > ZLibCompressionStrategy.Fixed)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
```

Comment:
I kind of liked what you used a few lines above. I wish there was a method that would let you provide both limits, like `ArgumentOutOfRangeException.ThrowIfNotWithin(min, max)`, but there isn't.

```suggestion
                ArgumentOutOfRangeException.ThrowIfLessThan(value, ZLibCompressionStrategy.Default);
                ArgumentOutOfRangeException.ThrowIfGreaterThan(value, ZLibCompressionStrategy.Fixed);
```

Reply:
Because `ZLibCompressionStrategy` is enum I was not able to use throw helpers

Reply:
Ah, okay. That's a bummer.

Reply:
You can, you just need to cast, eg
ArgumentOutOfRangeException.ThrowIfLessThan((int)value, (int) ZLibCompressionStrategy.Default, nameof(value));

Reply:
Yes, though didn't really think that was preferred, updating